crates/aptos-crypto/src/unit_tests/bls12381_test.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::iter::zip;
+use rand::distributions::Alphanumeric;
+use rand::prelude::IteratorRandom;
+use rand::Rng;
+use crate::{bls12381, Signature, SigningKey, test_utils::{KeyPair, TestAptosCrypto}, Uniform};
+use rand_core::OsRng;
+use crate::bls12381::bls12381_keys::{PrivateKey, PublicKey};
+use crate::bls12381::ProofOfPossession;
+
+/// Tests that an individual signature share computed correctly on a message m verifies on m.
+/// Tests that a signature share computed on a different message m' fails verification on m.
```

Comment:
Maybe this is a nit, but I liked the clarity of line 14. So, I suggest rewriting the line 13 as "...computed correctly on a message m passes verification on m."